### ISSUE TITLE
files/subfolder of ignored folders are not ignored

### ISSUE THREAD

**Issue by IvoDeK** [wangyucode/sftp-upload-action#23]
When ignoring it does ignore the main folder, but not the folders that are inside of it. This happens on either removing or uploading files.

#### Removing
![example](https://user-images.githubusercontent.com/59019739/221327456-d2488fa7-dbf7-4778-9288-c703f40790c9.png)
#### Uploading
![example2](https://user-images.githubusercontent.com/59019739/221327634-767461ac-aa2e-456b-b02e-654afda126f0.png)

**Comment by moazam1** [comment]
I have following files/folders excluded: `.git*,.DS_Store,vendor/,node_modules/,public\/assets` but yet it's still reading. It's taking about 12 minutes to push a one line change.

**Comment by mhimon** [comment]
Facing same issue -_-

**Comment by micaelmz** [comment]
same here, for me, this causes an 80% increase in the upload time it should normally take.

**Comment by mhimon** [comment]
I did a quick fix using this before deploying.

```
- name: Remvoe node_modules, .github, .git
        run: |
          rm -rf node_modules
          rm -rf .github
          rm -rf .git
```

**Comment by wangyucode** [comment]
published a new release `v2.0.3`, please reopen this issue when it happens again.
